pkg/routeagent_driver/handlers/ovn/gateway_route_handler.go

```diff
@@ -123,12 +131,24 @@ func (h *GatewayRouteHandler) TransitionToGateway() error {
 }
 
 func (h *GatewayRouteHandler) newGatewayRoute(endpoint *submarinerv1.Endpoint) *submarinerv1.GatewayRoute {
+	var famCIDRs []string
+
+	for _, cidr := range endpoint.Spec.Subnets {
+		if k8snet.IPFamilyOfCIDRString(cidr) == h.ipFamily {
+			famCIDRs = append(famCIDRs, cidr)
+		}
+	}
```

Comment:
We can actually use `cidr.ExtractSubnets` here.

```suggestion
	famCIDRs := cidr.ExtractSubnets(h.ipFamily, endpoint.Spec.Subnets)
```